src/toil/wdl/wdl_functions.py

```diff
@@ -596,3 +598,103 @@ def floor(i):
 def ceil(i):
     return math.ceil(i)
 
+
+def _get_temp_file_path(function_name, temp_dir=None):
+    """
+    Get a unique path with basename in the format of "{function_name}_{UUID}.tmp".
+    """
+
+    if not temp_dir:
+        temp_dir = os.getcwd()
+
+    # Cromwell uses the MD5 checksum of the content as part of the file name. We use a UUID instead
+    # for now, since we're writing line by line via a context manager.
+    # md5sum = hashlib.md5(content).hexdigest()
+    # name = f'{function_name}_{md5sum}.tmp'
+
+    name = f'{function_name}_{uuid.uuid4()}.tmp'
+    return os.path.join(temp_dir, 'execution', name)
+
+
+def write_lines(in_lines, temp_dir=None, file_store=None):
```

Comment:
Noted! Adding type hints definitely helps with readability, although, I'm not sure if we should remove the assert since Python doesn’t enforce the type. 

I doubt this would be an issue in most cases, but say if someone calls `write_lines()` with a `String` input instead of `Array[String]` in WDL, our runner would then split the string character by character instead of raising an error. 

Example workflow:
```wdl
workflow w {
  call t
}

task t {
  String in_str = "some string"

  command {
    cp ${write_lines(in_str)} output.txt
  }

 output {
    File out = 'output.txt'
 }
}
```
Cromwell fails with:
```
Caused by: common.exception.AggregatedMessageException: Error(s):
Could not evaluate expression: write_lines(in_str): Cannot cast wom.values.WomString to wom.values.WomArray$$anon$1
```
I might be overthinking this and the assertion error would be silently consumed anyway, but I intended to replicate Cromwell’s behavior as much as possible. Let me know what you think though, I’ll add type hints later today and I’m happy either way with the assert. :)

Reply:
Oh, that's a good catch!  Yes, I think the asserts are worth it then.

One more note then, you can add messages to asserts which might improve the readibility of the error when it occurs.

For example, instead of this:
```
var = 'abc'
assert isinstance(var, list)

# Traceback (most recent call last):
#   File "/home/quokka/delete/debug/toil/scrap.py", line 24, in <module>
#     assert isinstance('abc', list)
# AssertionError
```

It could be this:
```
var = 'abc'
assert isinstance(var, list), f'read_lines() requires "{var}" to be a list!  Not: {type(var)}'

# Traceback (most recent call last):
#   File "/home/quokka/delete/debug/toil/scrap.py", line 33, in <module>
#     assert isinstance(var, list), f'read_lines() requires "{var}" to be a list!  Not: {type(var)}'
# AssertionError: read_lines() requires "abc" to be a list!  Not: <class 'str'>
```

Reply:
Oh, that's a good catch!  Yes, I think the asserts are worth it then.

One more note then, you can add messages to asserts which might improve the readibility of the error when it occurs.

For example, instead of this:
```
var = 'abc'
assert isinstance(var, list)

# Traceback (most recent call last):
#   File "/home/quokka/delete/debug/toil/scrap.py", line 24, in <module>
#     assert isinstance('abc', list)
# AssertionError
```

It could be this:
```
var = 'abc'
assert isinstance(var, list), f'read_lines() requires "{var}" to be a list!  Not: {type(var)}'

# Traceback (most recent call last):
#   File "/home/quokka/delete/debug/toil/scrap.py", line 33, in <module>
#     assert isinstance(var, list), f'read_lines() requires "{var}" to be a list!  Not: {type(var)}'
# AssertionError: read_lines() requires "abc" to be a list!  Not: <class 'str'>
```